File: src/main/java/com/agenthub/domain/model/PermissionStrategy.java
package com.agenthub.domain.model;

import com.agenthub.domain.enums.AgentDataSourceProtocol;
import com.agenthub.domain.enums.OperationLevel;
import com.agenthub.domain.enums.TableOperation;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.Map;
import java.util.Set;

/**
 * 权限策略 - 第 5 个策略
 * <p>细粒度 CRUD 授权，区别于传统层级 READ_ONLY/READ_WRITE/ADMIN。</p>
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PermissionStrategy {
    private String id;
    private String tenantId;
    private String workspaceId;
    private String name;
    private String description;
    private Set<String> allowedRoles;
    private Set<OperationLevel> allowedOperations;
    private Set<AgentDataSourceProtocol> protocolBlocklist;
    private boolean dangerousSqlBlock;
    private Set<TableOperation> requireApprovalFor;
    private Map<String, Set<TableOperation>> tablePermissions;
    private int rateLimitPerMinute;
    private int rateLimitPerHour;
    private boolean auditLogEnabled;
    private int auditLogRetentionDays;
    private boolean piiMaskingOnResult;
    private Instant createdAt;
    private Instant updatedAt;
}

File: src/main/web/src/types/permission-strategy.ts
// 权限策略类型 - 与后端 PermissionStrategyResponse 对齐

export interface PermissionStrategy {
  id: string
  tenantId: string
  workspaceId: string
  name: string
  description?: string
  allowedRoles: string[]
  allowedOperations: string[]
  protocolBlocklist: string[]
  dangerousSqlBlock: boolean
  requireApprovalFor: string[]
  tablePermissions: Record<string, string[]>
  rateLimitPerMinute: number
  rateLimitPerHour: number
  auditLogEnabled: boolean
  auditLogRetentionDays: number
  piiMaskingOnResult: boolean
  createdAt?: string
  updatedAt?: string
}
